lib/livebook_web/live/home_live/edit_sessions_component.ex

```diff
@@ -0,0 +1,75 @@
+defmodule LivebookWeb.HomeLive.EditSessionsComponent do
+  use LivebookWeb, :live_component
+
+  import LivebookWeb.HomeLive.SessionListComponent, only: [toggle_edit: 1]
+
+  @impl true
+  def render(assigns) do
+    ~H"""
+    <div class="p-6 pb-4 flex flex-col space-y-8">
+      <h3 class="text-2xl font-semibold text-gray-800">
+        <%= title(@action) %>
+      </h3>
+      <.message action={@action} selected_sessions={@selected_sessions} sessions={@sessions}/>
+      <div class="mt-8 flex justify-end space-x-2">
+        <button class="button-base button-red" phx-click={toggle_edit(:off) |> JS.push(@action, target: @myself)}>
+          <.remix_icon icon="close-circle-line" class="align-middle mr-1" />
+          <%= button_label(@action) %>
+        </button>
+        <%= live_patch "Cancel", to: @return_to, class: "button-base button-outlined-gray" %>
+      </div>
+    </div>
+    """
+  end
+
+  defp message(%{action: "close_all"} = assigns) do
+    ~H"""
+    <p class="text-gray-700">
+      Are you sure you want to close <%= pluralize(length(@selected_sessions), "session", "sessions") %>?
+      <%= if not_persisted_count(@selected_sessions) > 0 do %>
+      <br/>
+        <span class="font-medium">Important:</span>
+        <%= pluralize(
+          not_persisted_count(@selected_sessions),
+          "notebook is not persisted and its content will be lost.",
+          "notebooks are not persisted and their content will be lost."
+        ) %>
+      <% end %>
+    </p>
+    """
+  end
+
+  defp message(%{action: "disconnect"} = assigns) do
+    ~H"""
+    <p class="text-gray-700">
+      Are you sure you want to disconnect <%= pluralize(length(@selected_sessions), "session", "sessions") %>?
+    </p>
+    """
+  end
+
+  @impl true
+  def handle_event("close_all", %{}, socket) do
+    socket.assigns.selected_sessions
+    |> Enum.map(&Livebook.Session.close(&1.pid))
```

Comment:
Since we don't care about the result:

```suggestion
    |> Enum.each(&Livebook.Session.close(&1.pid))
```